pkgs/by-name/ap/apkeditor/package.nix

```diff
@@ -0,0 +1,103 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  callPackage,
+
+  jre,
+  gradle_6,
+  makeWrapper,
+}:
+
+let
+  gradle = gradle_6;
+  REAndroidLibrary =
+    args:
+    let
+      inherit (args) pname version projectName;
+      outJar = "share/${projectName}/${projectName}.jar";
+      self = stdenv.mkDerivation (
+        {
+          __darwinAllowLocalNetworking = true;
+
+          buildInputs = [ jre ];
+          nativeBuildInputs = [ gradle ];
+
+          gradleFlags = [ "-Dfile.encoding=utf-8" ];
+          gradleBuildTask = "jar";
+          doCheck = true;
+
+          inherit outJar;
+          installPhase = ''
+            runHook preInstall
+            install -Dm644 build/libs/*.jar $out/${outJar}
+            runHook postInstall
+          '';
+        }
+        // args
+        // {
+          meta = {
+            sourceProvenance = with lib.sourceTypes; [
+              fromSource
+              binaryBytecode # mitm cache
+            ];
+          } // args.meta;
+        }
+      );
+    in
+    self;
+
+  arsclib = callPackage ./arsclib { inherit gradle REAndroidLibrary; };
+  smali = callPackage ./smali { inherit gradle REAndroidLibrary; };
+  jcommand = callPackage ./jcommand { inherit gradle REAndroidLibrary; };
+
+  apkeditor =
+    let
+      pname = "apkeditor";
+      version = "1.4.1";
+      projectName = "APKEditor";
+    in
+    REAndroidLibrary {
+      inherit pname version projectName;
+
+      # When you need to update **/deps.json for the dependencies (e.g. for smali),
+      # uncomment the following line and run `nix build apkeditor.smali.mitmCache.updateScript`.
+
+      # inherit arsclib smali jcommand;
+
+      src = fetchFromGitHub {
+        owner = "REAndroid";
+        repo = "APKEditor";
+        rev = "V${version}";
+        hash = "sha256-a72j9qGjJXnTFeqLez2rhBSArFVYCX+Xs7NQd8CY5Yk=";
+      };
+
+      nativeBuildInputs = [
+        gradle
+        makeWrapper
+      ];
+
+      gradleBuildTask = "fatJar";
+
+      preConfigure = ''
+        ln -sf ${arsclib}/${arsclib.outJar} libs/ARSCLib.jar
+        ln -sf ${smali}/${smali.outJar} libs/smali.jar
+        ln -sf ${jcommand}/${jcommand.outJar} libs/JCommand.jar
+      '';
```

Comment:
A commend would be nice describing why this needs to be done and how this works as this isn't a commonly understood pattern.